src/containerbuddy/etcd.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"time"
+
+	"encoding/json"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/etcd/client"
+)
+
+// Etcd is a service discovery backend for CoreOS etcd
+type Etcd struct {
+	Client client.Client
+	API    client.KeysAPI
+	Prefix string
+}
+
+// EtcdServiceNode is an instance of a service
+type EtcdServiceNode struct {
+	ID      string   `json:"id"`
+	Name    string   `json:"name"`
+	Address string   `json:"address"`
+	Port    int      `json:"port"`
+	Tags    []string `json:"tags,omitempty"`
+}
+
+// NewEtcdConfig creates a new service discovery backend for etcd
+func NewEtcdConfig(config map[string]interface{}) Etcd {
+	etcd := Etcd{
+		Prefix: "/containerbuddy",
+	}
+	etcdConfig := client.Config{}
+	switch endpoints := config["endpoints"].(type) {
+	case string:
+		etcdConfig.Endpoints = []string{endpoints}
+	case []string:
+		etcdConfig.Endpoints = endpoints
+	default:
+		log.Fatal("Must provide etcd endpoints")
+	}
+
+	prefix, ok := config["prefix"].(string)
```

Comment:
This will result in a panic if `prefix` isn't set: http://play.golang.org/p/v1uYdrWXAM